File: C++/Calculadora.cpp
#include <iostream>
#include <limits>
#include <cmath>

using namespace std;

class Calculator {
private:
    int operando1;
    int operando2;

public:
    Calculator() {}

    Calculator(int operando1, int operando2) {
        this->operando1 = operando1;
        this->operando2 = operando2;
    }

    int sumar() {
        return operando1 + operando2;
    }

    int restar() {
        return operando1 - operando2;
    }

    int multiplicar() {
        return operando1 * operando2;
    }

    double dividir() {
        if (operando2 != 0) {
            return static_cast<double>(operando1) / operando2;
        } else {
            cout << "SYNTAX ERROR (no existe la division para cero)." << endl;
            return numeric_limits<double>::quiet_NaN();
        }
    }
};

int main() {
    int num1, num2;

    do {
        cout << "Ingrese el primer numero: ";
        while (!(cin >> num1)) {
            cout << "Por favor, ingrese un numero entero: ";
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
        }
    } while (cin.fail());

    do {
        cout << "Ingrese el segundo numero: ";
        while (!(cin >> num2)) {
            cout << "Por favor, ingrese un numero entero: ";
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
        }
    } while (cin.fail());

    Calculator calculadora(num1, num2);

    char operacion;

    do {
        cout << "Ingrese la operacion que desea realizar (+, -, *, /): ";
        cin >> operacion;
        if (operacion != '+' and operacion != '-' and operacion != '*' and operacion != '/') {
            cout << "Por favor, ingrese una operacion valida." << endl;
        }
    } while (operacion != '+' and operacion != '-' and operacion != '*' and operacion != '/');

    switch (operacion) {
        case '+': {
            cout << "Suma: " << calculadora.sumar() << endl;
            break;
        }
        case '-': {
            cout << "Resta: " << calculadora.restar() << endl;
            break;
        }
        case '*': {
            cout << "Multiplicación: " << calculadora.multiplicar() << endl;
            break;
        }
        case '/': {
            double resultado = calculadora.dividir();
            if (!isnan(resultado)) {
                cout << "Division: " << resultado << endl;
            }
            break;
        }
        default:
            cout << "SYNTAX ERROR" << endl;
    }
}

File: 1ER PARCIAL/JAVA CODIGOS/CALCULADORA/Calculadora.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package com.mycompany.calculadora;

import java.util.Scanner;
import java.util.InputMismatchException;
/**
 *
 * @author USER
 */
public class Calculadora {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int num1, num2;

        do {
            try {
                System.out.println("Ingrese el primer número:");
                num1 = scanner.nextInt();
            } catch (InputMismatchException e) {
                System.out.println("Por favor, ingrese un número entero.");
                scanner.next(); // consume la entrada no válida
                num1 = Integer.MIN_VALUE; // Establecer un valor predeterminado
            }
        } while (num1 == Integer.MIN_VALUE);

        do {
            try {
                System.out.println("Ingrese el segundo número:");
                num2 = scanner.nextInt();
            } catch (InputMismatchException e) {
                System.out.println("Por favor, ingrese un número entero.");
                scanner.next(); // consume la entrada no válida
                num2 = Integer.MIN_VALUE; // Establecer un valor predeterminado
            }
        } while (num2 == Integer.MIN_VALUE);

        calculator calculadora = new calculator(num1, num2);

        char operacion;

        do {
            try {
                System.out.println("Ingrese la operación que desea realizar (+, -, *, /):");
                operacion = scanner.next().charAt(0);
                if (operacion != '+' && operacion != '-' && operacion != '*' && operacion != '/') {
                    throw new InputMismatchException();
                }
            } catch (InputMismatchException e) {
                System.out.println("Por favor, ingrese una operación válida.");
                scanner.next(); // consume la entrada no válida
                operacion = '\0'; // Establecer un valor predeterminado
            }
        } while (operacion == '\0');

        switch (operacion) {
            case '+' -> System.out.println("Suma: " + calculadora.sumar());
            case '-' -> System.out.println("Resta: " + calculadora.restar());
            case '*' -> System.out.println("Multiplicación: " + calculadora.multiplicar());
            case '/' -> System.out.println("División: " + calculadora.dividir());
            default -> System.out.println("SYNTAX ERROR");
        }
    }
}
